### ISSUE TITLE
Close ringbuffer properly on disconnect

### ISSUE THREAD

**Issue by julesdesmit** [dusk-network/dusk-blockchain#1053]
On a devnet run, this was observed:

```
{"error":"write tcp 157.245.99.89:7100-\u003e68.183.217.191:38116: use of closed network connection","level":"warning","msg":"error in writing to WriteCloser","process":"eventbus","queue":"ringbuffer","time":"2021-05-19T23:04:58Z"}
{"error":"ringbuffer is closed","level":"warning","msg":"ringbuffer closed","process":"eventbus","queue":"ringbuffer","time":"2021-05-19T23:04:58Z"}
{"count":21,"level":"debug","msg":"collected agreement","process":"agreement","quorum":16,"round":4730,"step":3,"time":"2021-05-19T23:04:58Z"}
{"level":"debug","msg":"quorum reached","process":"agreement","round":4730,"step":3,"time":"2021-05-19T23:04:58Z"}
{"level":"debug","msg":"received_2nd_step_reduction","process":"secondstep reduction","round":4730,"step":3,"time":"2021-05-19T23:04:58Z"}
{"level":"debug","msg":"quorum reached","process":"secondstep reduction","result_empty?":false,"round":4730,"step":3,"time":"2021-05-19T23:04:58Z"}
{"error":"rpc error: code = Canceled desc = context canceled","level":"warning","msg":"Invalid score message","process":"selector","time":"2021-05-19T23:04:58Z"}
{"error":"ringbuffer is closed","level":"warning","msg":"ringbuffer closed","process":"eventbus","queue":"ringbuffer","time":"2021-05-19T23:04:58Z"}
```

It seems, on disconnect, that the ring buffer doesn't close properly in all cases.

**Comment by goshawk-3** [comment]
Most probably a corner case that need to be reproduced and handled.

**Comment by goshawk-3** [comment]
This has been recently addressed with #1241
